### PR TITLE
TVAULT-4643: unable to install horizon component after openstack upgrade

### PR DESCRIPTION
**Pull request by rohit-keshri**
image

### CODE REVIEW

File: ansible/roles/ansible-horizon-plugin/tasks/collect_compress.yml [hunk 1]
```diff
@@ -39,9 +39,33 @@
...
+  set_fact:
+    file_names: "{{ os_local_settings_path.stdout.split('\n')}}"
+  when: os_local_settings_path.stdout != ""
+
+- set_fact:
+    os_local_settings_path_length: "{{ file_names|length }}"
+  when: os_local_settings_path.stdout != ""
+
 - name: HORIZON_CONFIG
   shell: . /openstack/venvs/horizon*/bin/activate && echo "HORIZON_CONFIG['customization_module'] = 'dashboards.overrides'" > `echo $os_local_settings_path | cut -d ' ' -f1 |{ read trilio_setting_path ; echo $trilio_setting_path/_001_trilio_dashboard.py ; }`
```
Comment (abhijeetpatra) [inline ansible/roles/ansible-horizon-plugin/tasks/collect_compress.yml:52]: this path may not be correct

Comment (rohit-keshri) [reply ansible/roles/ansible-horizon-plugin/tasks/collect_compress.yml]: now we are taking horizon virtual environment path from user.

File: ansible/roles/ansible-horizon-plugin/tasks/collect_compress.yml [hunk 2]
```diff
@@ -36,12 +36,20 @@
...
+  shell: find / -type d -path /sys -prune -o -name "*local_settings.d" 2>/dev/null | grep -E 'openstack_dashboard'
   register: os_local_settings_path
 
 - name: HORIZON_CONFIG
-  shell: . /openstack/venvs/horizon*/bin/activate && echo "HORIZON_CONFIG['customization_module'] = 'dashboards.overrides'" > `echo $os_local_settings_path | cut -d ' ' -f1 |{ read trilio_setting_path ; echo $trilio_setting_path/_001_trilio_dashboard.py ; }`
-  when: os_local_settings_path != ""
+  shell: echo "HORIZON_CONFIG['customization_module'] = 'dashboards.overrides'" > `echo {{os_local_settings_path.stdout}} | cut -d ' ' -f1 |{ read trilio_setting_path ; echo $trilio_setting_path/_001_trilio_dashboard.py ; }`
+  when:
+    - os_local_settings_path != ""
+    -
```
Comment (DeepikaAFour) [inline ansible/roles/ansible-horizon-plugin/tasks/collect_compress.yml:46]: is there a condition here?

Comment (rohit-keshri) [reply ansible/roles/ansible-horizon-plugin/tasks/collect_compress.yml]: no any condition there , "-" removed .

**Comment by abhijeetpatra** [comment]
closing this PR as it would be fixed in 4.2

**Pull request closed without merging**